tests/functional/test_install_config.py

```diff
@@ -415,3 +415,44 @@ def get_credential(url, username):
         assert "get_credential was called" in result.stderr
     else:
         assert "get_credential was called" not in result.stderr
+
+
+def test_prioritize_url_credentials_over_netrc(
+    script: PipTestEnvironment,
+    data: TestData,
+    cert_factory: CertFactory,
+) -> None:
+    cert_path = cert_factory()
+    ctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    ctx.load_cert_chain(cert_path, cert_path)
+    ctx.load_verify_locations(cafile=cert_path)
+    ctx.verify_mode = ssl.CERT_REQUIRED
+
+    server = make_mock_server(ssl_context=ctx)
+    server.mock.side_effect = [
+        package_page(
+            {
+                "simple-3.0.tar.gz": "/files/simple-3.0.tar.gz",
+            }
+        ),
+        authorization_response(str(data.packages / "simple-3.0.tar.gz")),
+    ]
+
+    url = f"https://USERNAME:PASSWORD@{server.host}:{server.port}/simple"
+
+    netrc = script.scratch_path / ".netrc"
+    netrc.write_text(f"machine {server.host} login username password password")
```

Comment:
Can we add a `wrong` to each of these -- login wrongusername password wrongpassword -- to make it clearer that these aren't being used and to avoid relying on a case-sensitive compare (I know, it's username/password -- but it'll be clearer).